WcaOnRails/app/models/team_member.rb

```diff
@@ -0,0 +1,20 @@
+class TeamMember < ActiveRecord::Base
+  belongs_to :team
+  belongs_to :user
+
+  attr_accessor :current_user
+
+  validate :cannot_demote_leader
+  def cannot_demote_leader
+    if end_date != nil && end_date < Date.today && team_leader
+      errors.add(:end_date, "A team leader must be a current member.")
+    end
+  end
+
+  validate :cannot_demote_oneself
+  def cannot_demote_oneself
+    if current_user == self.user_id && end_date != nil && end_date < Date.today
```

Comment:
Very clean validations!

I see you have this `end_date != nil && end_date < Date.today` in 2 places. How about pulling it out into a `def current_member?` or `def ongoing_member?` method? I feel like that would be easier to read.